libraries/botbuilder-dialogs-adaptive/src/functions/isDialogActiveFunction.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    Expression,
+    ExpressionEvaluator,
+    MemoryInterface,
+    Options,
+    ReturnType,
+    ValueWithError,
+} from 'adaptive-expressions';
+
+/**
+ * Defines isDialogActive(id) expression function.
+ * This expression will return true if any of the dialog ids is on the dialog execution stack.
+ *
+ * @example isDialogActive('dialog1')
+ * @example isDialogActive('dialog1', 'dialog2', 'dialog3')
+ */
+export class IsDialogActiveFunction extends ExpressionEvaluator {
+    /**
+     * Function identifier name.
+     */
+    public static readonly functionName = 'isDialogActive'; // `name` is reserved in JavaScript.
+
+    /**
+     * Function identifier alias.
+     */
+    public static readonly functionAlias = 'isActionActive';
+
+    /**
+     * Intializes a new instance of the [IsDialogActiveFunction](xref:botbuilder-dialogs-adaptive.IsDialogActiveFunction) class.
+     */
+    public constructor() {
+        super(IsDialogActiveFunction.functionName, IsDialogActiveFunction.function, ReturnType.Boolean);
+    }
+
+    private static function(expression: Expression, state: MemoryInterface, options: Options): ValueWithError {
+        const stack: string[] = state.getValue('dialogcontext.stack');
+        if (!stack) {
+            return { value: undefined, error: 'dialogcontext.stack not found' };
+        }
+
+        const args: string[] = [];
+        expression.children.forEach((child) => {
+            const { value, error } = child.tryEvaluate(state, options);
+            if (error) {
+                return { value: undefined, error };
```

Comment:
This `return` isn't captured anywhere since it's inside a `forEach`. Perhaps wrap the `forEach` inside a try-catch and throw an error, or use a `for` loop.

Reply:
Thanks for reminding me, wrapped by a try-catch instead.